pkg/server/server.go

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package server
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"syscall"
+
+	"github.com/go-logr/logr"
+	"github.com/oklog/run"
+
+	"github.com/NVIDIA/knavigator/pkg/config"
+	"github.com/NVIDIA/knavigator/pkg/engine"
+)
+
+type Server struct {
+	s   *http.Server
+	log *logr.Logger
+}
+
+type TaskHandler struct {
+	eng *engine.Eng
+}
+
+type ResetHandler struct {
+	eng *engine.Eng
+}
+
+func New(log *logr.Logger, eng *engine.Eng, port int) *Server {
+	mux := http.NewServeMux()
+	mux.Handle("/task", &TaskHandler{eng: eng})
+	mux.Handle("/reset", &ResetHandler{eng: eng})
+	return &Server{
+		log: log,
+		s: &http.Server{ // #nosec G112
+			Addr:    fmt.Sprintf(":%d", port),
+			Handler: mux,
+		},
+	}
+}
+
+func (srv *Server) Run() error {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	var g run.Group
+	// Signal handler
+	g.Add(run.SignalHandler(ctx, os.Interrupt, syscall.SIGTERM))
+	// Server
+	g.Add(
+		func() error {
+			srv.log.Info("Starting server", "address", srv.s.Addr)
+			return srv.s.ListenAndServe()
```

Comment:
Do we need some form of authn and authz? I think a user who is able to access the server will be able to privilege escalate since the user and the server use two different RBAC means. What are your thoughts on this?

Reply:
I didn't want to complicate things. This is a local kind cluster, the nodes are virtual. I think authn/authz will be an overkill.

Reply:
Yeah, fair point. Then we should add documentation describing the risks for users who want to have a long running cluster, maybe for simulation testing/QA. 

I was also wondering; can we leverage the JWT tokens generated for Kubernetes service accounts? Use the tokens for a service account (https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/#manually-create-an-api-token-for-a-serviceaccount) for making calls to the Kubernetes API and let the Kubernetes API server handle auth. User will provide the token as a part of the Authorization header.